### PR TITLE
Improve grammar and writing in Pipeline compilations page

### PR DESCRIPTION
**Pull request by Calinou**
- Mention example of off-screen rendering using ColorRect to hide the root viewport's 3D rendering or a SubViewport node.
- Split into more paragraphs to improve readability.
- Link to class reference pages when relevant.
- Use sentence case for headings to match the rest of the documentation.
- Wrap source file to 80 characters like other pages.

### CODE REVIEW

**Comment by mhilbrunner** [comment]
> Wrap source file to 80 characters like other pages.

For the future, this is probably worth doing in a separate commit, because it makes reviewing changes in Github's diff painful, even when set to ignore whitespace.

**Review by mhilbrunner — APPROVED** [review]

File: tutorials/performance/pipeline_compilations.rst [hunk 1]
```diff
@@ -1,31 +1,84 @@
...
+Pipeline compilation, also commonly known as shader compilation, is an expensive
+operation required by the engine to be able to draw any kind of content with the
+GPU.
+
+In more precise terms, *shader compilation* involves the translation of the GLSL
+code that Godot generates into an intermediate format that can be shared across
+systems (such as SPIR-V when using Vulkan). However, this format can't be used
+by the GPU directly.
+
+*Pipeline compilation* is the step the GPU driver performs where it'll convert
```
Comment (tetrapod00) [inline tutorials/performance/pipeline_compilations.rst:15]:
```suggestion
*Pipeline compilation* is the step the GPU driver performs where it converts
```
or
```suggestion
*Pipeline compilation* is the step where the GPU driver converts
```

**Review by tetrapod00 — APPROVED** [review]
Suggested change is optional, the rest (still) looks good

**Comment by mhilbrunner** [comment]
Thanks!

**Pull request merged**